sb_advanced_nextbots/lua/entities/sb_advanced_nextbot_base/enemy.lua

```diff
@@ -225,6 +225,7 @@ end
 --]]------------------------------------
 function ENT:ShouldBeEnemy(ent)
 	if ent:IsPlayer() and GetConVar("ai_ignoreplayers"):GetBool() then return false end
+	if ent:IsPlayer() and ent:IsFlagSet(FL_NOTARGET) then return false end

```

Comment:
No `ent:IsPlayer()` check is needed, `ent:IsFlagSet` can be used on any entity. Also, it will fully match the request title.